### ISSUE TITLE
Test of Develop Issuing CA alias

### ISSUE THREAD

**Issue by tedescn** [certnanny/sscep#122]
@gotthardp, not sure if you received notifications to #111 testing output? I attached observations to that thread which is closed. 

Testing was partially successful, where the difficult stuff works:-) , but the i'm experiencing a variable truncation issue.

Regards
Nigel

**Comment by gotthardp** [comment]
Thanks, Nigel! I saw it, but I forgot to respond, sorry.

**Comment by gotthardp** [comment]
Hi again, it should be fixed now.

**Comment by gotthardp** [comment]
Concerning the other issue your mentioned, GetCACaps: it is meant to be automatic. The capabilities are queried before every request to automatically select the right options to use. Right now it is used only to decide between GET and POST, but it could be extended also to select the algorithms, etc.

**Comment by tedescn** [comment]
@gotthardp Just to confirm fix was successful. I just retested on CentOS 8 the GET method transaction (GetCACaps, GetCA, PKIOperation{PKCSReq, CertPoll}). For your knowledge, these requests are against the same [ VeriSign | Symantec | DigiCert ]  OnSite Processing Centre, indicated on the README.md.

I will investigate both the extension of this legacy capabiilty, and the emerging next generation SCEP capability.

Thanks

Nigel

**Comment by tedescn** [comment]
@gotthardp, WRT comment of GetCACaps could be used to select algorithms, this is something I do today. Essentially, I write the response to file, and parse the values into system variables. These variables are used on the sscep command line.

I hope this is useful information.

Regards
Nigel

**Comment by gotthardp** [comment]
And you obtain the GetCACaps values via some other tool, or you made a patch, or how?

**Comment by tedescn** [comment]
You are correct, some other tool. weither:

1. wget ${SCEP_RESPONDER}"?operations=GetCACaps""&message="${IssuingCA_Identifier} -O ${CACAPS}, or
2. wget ${SCEP_RESPONDER}"?operations=GetCACaps" -O ${CACAPS}

I then pull the values from the output file.

**Comment by gotthardp** [comment]
BTW, the getcaps operation works, it is just not documented :)
```
user@server:~/projects/sscep$ ./sscep getcaps -u http://192.168.1.1/scep
./sscep: scep capabilities: DES3, POSTPKIOperation, Renewal, SHA-1, SHA-224, SHA-256, SHA-384, SHA-512
```

**Comment by tedescn** [comment]
Thanks.

I checked the same yesterday, with similar confirmation. I also need to to talk to our platform operators regards what happens backend regards the impact of the missing "...message=${IssuingCA_Identifier}..."?

I do understand the SSCEP implementation reflects the the RFC8894.

**Comment by gotthardp** [comment]
Well, the SSCEP implementation has been made by a lot of people through the years, so it may not always reflect the latest RFC. When it does not reflect the RFC 8894, or when its operation is causing some troubles, please open an issue for that.

**Comment by gotthardp** [comment]
The variable truncation has been addressed in 0f7b175.
